### PR TITLE
Compensate AK09916 on CubeOrange for magnetic impact of heater

### PR DESCRIPTION
**Pull request by tridge**
This adds an offset to the internal AK09916 compass based on the heater duty cycle. It is needed as the heater produces a strong magnetic field.
This still leaves the internal compass as quite noisy, but the average is good which is what really matters for the EKF
Without correction:
image
with correction:
image

### CODE REVIEW

File: libraries/AP_BoardConfig/AP_BoardConfig.h [hunk 1]
```diff
@@ -189,6 +189,9 @@ class AP_BoardConfig {
 
 #if HAL_HAVE_IMU_HEATER
     void set_imu_temp(float current_temp_c);
+    float get_heater_duty_cycle(void) {
```
Comment (peterbarker) [inline libraries/AP_BoardConfig/AP_BoardConfig.h:192]:
```suggestion
    float get_heater_duty_cycle(void) const {
```

File: libraries/AP_HAL_ChibiOS/hwdef/CubeOrange/hwdef.dat [hunk 2]
```diff
@@ -281,6 +281,9 @@ COMPASS AK8963:probe_mpu9250 1 ROTATION_YAW_270
 # compass as part of ICM20948 on newer cubes
 COMPASS AK09916:probe_ICM20948 0 ROTATION_ROLL_180_YAW_90
 
+# offset the internal compass for EM impact of the IMU heater
```
Comment (peterbarker) [inline libraries/AP_HAL_ChibiOS/hwdef/CubeOrange/hwdef.dat:284]:
```suggestion
# offset the internal compass for EM impact of the IMU heater (body frame milligauss)
```

**Review by peterbarker — CHANGES_REQUESTED** [review]
LGTM apart from the `const`.

File: libraries/AP_BoardConfig/AP_BoardConfig.h (hunk 1 above)
Comment (rmackay9) [inline libraries/AP_BoardConfig/AP_BoardConfig.h:192]: as mentioned on the dev call, let's add a comment to clarify the units of the method.

**Review by rmackay9 — APPROVED** [review]

**Pull request merged**